src/ui/Components/ConverterPanel.ts

```diff
@@ -75,6 +84,9 @@ export class ConverterPanelModel {
         var u32le = Converter.numConv(data, 4, false, false);
         this.unixts = u32le ? dateFormat(new Date(parseInt(u32le) * 1000), "yyyy-mm-dd HH:MM:ss") : "";
 
+        var u64le = Converter.numConv(data, 8, false, false);
+        this.filetime = u64le ? dateFormat(new Date(Converter.fileToUnixTime(parseInt(u64le)) * 1000), "yyyy-mm-dd HH:MM:ss") : "";
```

Comment:
@sealmove Unfonutately, as all [numbers in JavaScript ](https://developer.mozilla.org/en-US/docs/Glossary/Number) are `double`s with just 53 bits of mantissa, a 64-bit integer will lose precision during the conversion to it (using the `parseInt` method). It means that not all integers larger than **2<sup>53</sup> - 1** (available in the `Number.MAX_SAFE_INTEGER` constant) can't be represented accurately in the native JavaScript number type: e.g. `Number.MAX_SAFE_INTEGER + 1 == Number.MAX_SAFE_INTEGER + 2`, see [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER).

But it shouldn't happen here, because the [`Converter.numConv`](https://github.com/kaitai-io/kaitai_struct_webide/blob/bbb8b8866994b2fad75072991453dec553076895/src/ui/Components/ConverterPanel.ts#L17) method works internally with a [`bigInt`](https://www.npmjs.com/package/big-integer#bigintnumber-base-alphabet-casesensitive), though it returns a `string` (don't know why). Anyway, I would suggest to convert the string to the [`bigInt`](https://www.npmjs.com/package/big-integer#bigintnumber-base-alphabet-casesensitive) type, and do the arithmetics you need to do in the `fileToUnixTime` method using [its methods](https://www.npmjs.com/package/big-integer#methods) instead of the native arithmetic operators.

And I just checked that [`Date` object has more than sufficient range](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date#The_ECMAScript_epoch_and_timestamps) for not losing precision here, it's even far larger than `FILETIME` has (maximum year in `FILETIME` is [~30828](https://stackoverflow.com/a/18188484), maximum year of `Date` is [275760](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date#The_ECMAScript_epoch_and_timestamps)), so there shouldn't be a problem, if you choose a safe method of initializing it.

Reply:
> there shouldn't be a problem, if you choose a _safe method_ of initializing it.

@sealmove Actually, you can _safely_ ignore that "safe method". I've just tested [on RunKit](https://runkit.com/embed/6e7ab3oi98n7) the maximum year 30828 of `FILETIME`, and the number that represents that `Date` (number of milliseconds that have elapsed since midnight on January 1, 1970, UTC) is still ~10 times less than the `Number.MAX_SAFE_INTEGER`, so it will be fine to just stuff that `Unix timestamp * 1000` number into the `new Date()` constructor, after you do the arithmetic with `bigInt`.